test/functional-api/conversions/conversion.it-spec.ts

```diff
@@ -0,0 +1,405 @@
+
+    // challange data
+
+    const chalDataCommunity = chalRes.body.data.hub.challenge.community;
+    const chalDataContext = chalRes.body.data.hub.challenge.context;
+    const chalDataAgent = chalRes.body.data.hub.challenge.agent;
+    const chalDataApplication = chalRes.body.data.hub.challenge.application;
+    const chalDataAuthorization = chalRes.body.data.hub.challenge.authorization;
+    const chalDataChallenges = chalRes.body.data.hub.challenge.challenges;
+    const chalDataOpportunities = chalRes.body.data.hub.challenge.opportunities;
+    const chalDataPreferences = chalRes.body.data.hub.challenge.preferences;
+    const chalDataTagset = chalRes.body.data.hub.challenge.tagset;
+    const chalDataTemplates = chalRes.body.data.hub.challenge.templates;
+    const chalDataLeadOrg =
+      chalRes.body.data.hub.challenge.community.leadOrganizations;
+    const chalDataNameId = chalRes.body.data.hub.challenge.nameID;
+    const chalDataDisplayName = chalRes.body.data.hub.challenge.displayName;
+
+    // create Opportunity
+    const resOpp = await mutation(
+      createOpportunity,
+      opportunityVariablesData(
+        opportunityName,
+        `success-oppnameid${uniqueId}`,
+        newChallId
+      )
+    );
+
+    const newOppId = resOpp.body.data.createOpportunity.id;
+    const newOppCommunityId = resOpp.body.data.createOpportunity.community.id;
+    const assignOpportunityOrgLead = await assignOrganizationAsCommunityLeadFunc(
+      newOppCommunityId,
+      entitiesId.organizationId
+    );
+
+    await assignUserAsCommunityMemberFunc(newOppCommunityId, users.hubMemberId);
+    await assignUserAsCommunityLeadFunc(newOppCommunityId, users.hubMemberId);
+    const oppRes = await getOpportunityData(entitiesId.hubId, newOppId);
+
+    // opportunity data
+
+    const oppDataCommunity = oppRes.body.data.hub.opportunity.community;
+    const oppDataContext = oppRes.body.data.hub.opportunity.context;
+    const oppDataAgent = oppRes.body.data.hub.opportunity.agent;
+    const oppDataApplication = oppRes.body.data.hub.opportunity.application;
+    const oppDataAuthorization = oppRes.body.data.hub.opportunity.authorization;
+    // const oppDataChallenges = oppRes.body.data.hub.opportunity.challenges;
+    // const oppDataOpportunities = oppRes.body.data.hub.opportunities;
+    const oppDataPreferences = oppRes.body.data.hub.opportunity.preferences;
+    const oppDataTagset = oppRes.body.data.hub.opportunity.tagset;
+    const oppDataTemplates = oppRes.body.data.hub.opportunity.templates;
+    console.log(oppRes.body.data.hub.opportunity.community);
+    const oppDataLeadOrg =
+      oppRes.body.data.hub.opportunity.community.leadOrganizations;
+    const oppDataNameId = oppRes.body.data.hub.opportunity.nameID;
+    const oppDataDisplayName = oppRes.body.data.hub.opportunity.displayName;
+
+    // Act
+    const res = await convertChallengeToHub(newChallId);
+    console.log(res.body);
+
```

Comment:
Please save res.body.data.convertChallengeToHub in a separate variable so that latter code becomes more readable.